rpc/src/v1/impls/light/parity.rs

```diff
@@ -390,14 +390,14 @@ impl Parity for ParityClient {
 		})
 	}
 
-	fn block_header(&self, number: Trailing<BlockNumber>) -> BoxFuture<RichHeader> {
+	fn block_header(&self, number: Trailing<BlockNumber>) -> BoxFuture<Result<RichHeader>> {
```

Comment:
This is redundant -- Future can already have an error value. I recommend keeping the return type as is and changing the `map` on line 432 to an `and_then`.

Reply:
You are right, and your suggestion works great. @debris reverting the return type change and going with @jimpo 's suggestion! :)